wordpress-mobile/WordPress-iOS
# Add "Jetpack powered" badge to Notification Settings

Fixes #NA

This PR adds a "Jetpack powered" badge below the "Your Sites" section, in Notification Settings.

Given that the changes only impacted `NotificationSettingsViewController`, and that the same had most of the code written between 2015 and 2016, I also took the occasion to modernize it a bit. There is still stuff that can be done, but did not want this PR to diverge.

iPad dark | iPhone light
-- | --
<img width="375" src="https://user-images.githubusercontent.com/34376330/180238154-5c64fdf5-c26b-466c-8ac9-fe60820e5404.png"> | <img width="300" src="https://user-images.githubusercontent.com/34376330/180238155-414367af-8fc3-4d5f-9f60-058054b22cd9.png">

**KNOWN ISSUES: if you have many sites, the animation that expands and collapses the cells will generate a bit of funky movement of the badge. Since having many sites is an edge case, we decided that we can safely ship and then address the animation issue separately.**

To test:
- Build/run the `WordPress` target from this branch
- Go to Notifications then tap the cog in the top right to access Notification settings
- Make sure the "Jetpack powered" badge appears below the "Your Sites" section
- Stop the execution and build the `Jetpack` target
- Make sure you don't see the badge
- Stop the execution, change the `.jetpackPowered` Feature flag to false
- Build the `WordPress` target again and make sure the badge does not appear

## Regression Notes
1. Potential unintended areas of impact
`NotificationSettingsViewController`

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Test on a variety of devices and simulators

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:
